### ISSUE TITLE
csc.dll is confused for source file under dotnet test

### ISSUE THREAD

**Issue by felix-b** [daveaglick/Buildalyzer#73]
### Problem

I use Buildalyzer to populate an AdhocWorkspace, like this:
```csharp
var workspace = new AdhocWorkspace();
var analyzerManager = new AnalyzerManager();

foreach (var filePath in projectFilePaths)
{
    ProjectAnalyzer analyzer = _analyzerManager.GetProject(filePath);
    analyzer.AddToWorkspace(workspace);
}
```
then I compile a project like this:
```csharp
var compilation = workspace.CurrentSolution.Projects.First().GetCompilationAsync().Result;
```
My integration tests (written with NUnit) successfully pass when run by ReSharper in Visual Studio. When I try to run them in a console with `dotnet test`, I get a weird behavior: GetCompilationAsync() produces diagnostics where the compiler complains:
```
C:\Program Files\dotnet\sdk\2.1.302\Roslyn\bincore\csc.dll(1,1): error CS0116: A namespace cannot directly contain members such as fields or methods
```
and several screens of strings similar to:
```
\u0013\u0015Microsoft Corporation1&0$\u0006\u0003U\u0004\u0003\u0013\u001dMicrosoft Time-Stamp PCA 2010\u0002\u00133\u0000
```

### Environment
```
> dotnet --info

.NET Core SDK (reflecting any global.json):
 Version:   2.1.302
 Commit:    9048955601

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.302\

Host (useful for support):
  Version: 2.1.2
  Commit:  811c3ce6c0

.NET Core SDKs installed:
  2.1.2 [C:\Program Files\dotnet\sdk]
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.201 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.302 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
```

**Comment by felix-b** [comment]
After examining MSBuild output, I see that CSC command line under `dotnet test` is different. Under ReSharper, it is:
```
C:\Program Files\dotnet\sdk\2.1.302\Roslyn\bincore\csc.dll /noconfig /unsafe- ........
```
whereas under `dotnet test` it is:
```
C:\Program Files\dotnet\dotnet.exe "C:\Program Files\dotnet\sdk\2.1.302\Roslyn\bincore\csc.dll" /noconfig /unsafe- ............
```
and therefore MSBuild output items include csc.dll in CscCommandLineArgs output item:
```
    Output Item(s): 
        CscCommandLineArgs=
            C:\Program Files\dotnet\sdk\2.1.302\Roslyn\bincore\csc.dll
            /noconfig
            /unsafe-
            .........
            HelloPage.cs
            Services\IGreetingService.cs
            .........
```
and accordingly, I get "C:\Program Files\dotnet\sdk\2.1.302\Roslyn\bincore\csc.dll" in the first item returned by `ProjectAnalyzer.GetSourceFiles()`

I guess that list of source files is retrieved from CscCommandLineArgs. Then excluding file paths with "dll" extension should do the work. Am I on the right path?

**Comment by daveaglick** [comment]
> I guess that list of source files is retrieved from CscCommandLineArgs. Then excluding file paths with "dll" extension should do the work. Am I on the right path?

That's exactly correct. Buildalyzer looks at what MSBuild told it the `CscCommandLineArgs` property is. Are you on the latest version? This is a known issue and there's already code to deal with it when using `AnalyzerResult.GetSourceFiles()`: https://github.com/daveaglick/Buildalyzer/blob/master/src/Buildalyzer/AnalyzerResult.cs#L52

**Comment by daveaglick** [comment]
I just published Buildalyzer 2.0 which runs MSBuild out-of-process (as opposed to the old way of running it via MSBuild APIs). In general, if you can build it Buildalyzer should now be able to build it. Given how drastically different this new technique is, I'm closing all existing bug reports.

Please try the new version and if you're still having problems, please open a new issue.
